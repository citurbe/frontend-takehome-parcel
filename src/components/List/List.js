import React from 'react';
import Gem from '../Gem/Gem';

const List = props => {
  return (
    <div>
      {props.gemList.map(gemData => (
        <Gem
          toggleSave={props.toggleSave}
          key={gemData.name}
          isSaved={props.savedGems.find(gem => gem.name === gemData.name)}
          {...gemData}
        />
      ))}
    </div>
  );
};

export default List;
