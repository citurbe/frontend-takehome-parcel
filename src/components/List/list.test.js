import React from 'react';
import List from './List';
import renderer from 'react-test-renderer';

const gemList = [
  {
    name: 'gem 1',
    version: '2.0',
    project_uri: 'http://test1.org'
  },
  {
    name: 'gem 2',
    version: '5.6',
    project_uri: 'http://test2.org'
  },
  {
    name: 'gem 3',
    version: '1.9.2',
    project_uri: 'http://test3.org'
  }
];

const savedGems = [
  {
    name: 'gem 2',
    version: '5.6',
    project_uri: 'http://test2.org'
  }
];

test('List renders correctly', () => {
  const component = renderer.create(
    <List gemList={gemList} savedGems={savedGems} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
