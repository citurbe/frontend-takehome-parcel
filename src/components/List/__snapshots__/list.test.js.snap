// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <div
    className="gem-card"
  >
    <a
      href="http://test1.org"
      target="blank"
    >
      gem 1
    </a>
    <p>
      version: 
      2.0
    </p>
    <button
      className="save-button"
      onClick={[Function]}
    >
      Save
    </button>
  </div>
  <div
    className="gem-card"
  >
    <a
      href="http://test2.org"
      target="blank"
    >
      gem 2
    </a>
    <p>
      version: 
      5.6
    </p>
    <button
      className="save-button"
      onClick={[Function]}
    >
      Un-Save
    </button>
  </div>
  <div
    className="gem-card"
  >
    <a
      href="http://test3.org"
      target="blank"
    >
      gem 3
    </a>
    <p>
      version: 
      1.9.2
    </p>
    <button
      className="save-button"
      onClick={[Function]}
    >
      Save
    </button>
  </div>
</div>
`;
